src/js/helpers/csvHelpers.js

```diff
@@ -6,96 +6,124 @@ import csv from 'csv';
+        if (fs.existsSync(indexPath)) {
+          try {
+            const items = fs.readJsonSync(indexPath);
+            items.forEach(item => {
+              item.category = category; // adds the category as a field since this is being flattened from the directory
+              index.push(item);
+            });
+          } catch (e) {
+            console.error(`Failed to read the category index at ${indexPath}`, e);
+          }
+        }
+      });
+    }
+  } catch (e) {
+    console.error(`Failed to read tool category directories for ${toolName}`, e);
+  }
+  return index;
 }
 
 /**
  * @description - combines all data needed for csv
  * @param {object} data - the data that the rest appends to
  * @param {object} contextId - to be merged in
- * @param {object} indexObject - to be used to get groupName
- * @param {array} indexObject - Array of index.json with {id, name} keys
  * @param {string} username
  * @param {timestamp} timestamp to be converted into date and time
- * @param {boolean} - This is a temporary flag to hide bad data
+ * @param {function} translate - the translation function
+ * @param {boolean} isTest - for testing purposes
  * @return {object}
  */
-export function combineData(data, contextId, username, timestamp) {
-  const flatContextId = flattenContextId(contextId);
+export function combineData(data, contextId, username, timestamp, translate, isTest = false) {
+  const flatContextId = flattenContextId(contextId, translate, isTest);
   const userTimestamp = userTimestampObject(username, timestamp);
-  const combinedData = Object.assign({}, data, flatContextId, userTimestamp);
-  return combinedData;
+  return Object.assign({}, data, flatContextId, userTimestamp);
 }
 /**
  * @description - flattens the context id for csv usage
  * @param {object} contextId - contextID object that needs to go onto the csv row
- * @param {string} isTest - for unit tests purposes
+ * @param {function} translate - translation function
+ * @param {boolean} isTest - for unit tests purposes
  * @return {object}
  */
-export const flattenContextId = (contextId, isTest) => {
-  const flatContextId = {
+export const flattenContextId = (contextId, translate, isTest = false) => {
+  return {
     tool: contextId.tool,
+    type: groupCategoryTranslated(contextId, translate, isTest),
     groupId: contextId.groupId,
     groupName: groupName(contextId, isTest),
     occurrence: contextId.occurrence,
-    quote: contextId.quote,
+    quote: flattenQuote(contextId.quote),
+    glQuote: contextId.glQuote,
+    occurrenceNote: contextId.occurrenceNote,
     bookId: contextId.reference.bookId,
     chapter: contextId.reference.chapter,
     verse: contextId.reference.verse
   };
-  return flatContextId;
+};
+
+/**
+ * Flattens a quote which may be an array of words
+ * @param {*} quote
+ * @returns {string}
+ */
+export const flattenQuote = quote => {
+  if (Array.isArray(quote)) {
+    quote = quote.map(item => item.word).join(" ");
```

Comment:
How are you handling punctuation ? it looks like you're not

Reply:
Good point. Do we already have a function that does this?

Reply:
Fixing

Reply:
Fixed